boxes/__init__.py

```diff
@@ -1771,7 +1632,335 @@ def fillHoles(self, pattern, border, max_radius, hspace=3, bspace=0, min_radius=
                     else:
                         raise ValueError("fillHoles - unknown hole style: %s)" % style)
                     # rinse and repeat
+        
+        elif pattern == "hex":
+            # use 'optimum' hole size
+            max_radius = min(max_radius_x, max_radius_y)
+            # set starting point
+            x = min_x + bspace - max_radius - hspace + offset_x
+            y = min_y + bspace + max_radius + offset_y 
+            while True:
+                # compute next point
+                x += 2 * max_radius + hspace
+                if x > max_x - bspace:
+                    row += 1
+                    if row % 2 == 0:
+                        x = min_x + max_radius + bspace + offset_x
+                    else:
+                        x = min_x + max_radius * 2 + hspace / 2 + bspace + offset_x
+                    y += math.sqrt(3) * (max_radius + hspace)
+                    if y > max_y - bspace:
+                        return
+                pt = Point(x, y).buffer(min_radius + bspace)
+                # check if point is within border
+                if borderPoly.contains(pt):
+                    pt1 = Point(x, y)
+                    # compute distance between hole and border
+                    r = borderPoly.exterior.distance(pt1) - bspace
+                    # if too small, dismiss
+                    if r < min_radius:
+                        continue
+                    # if too large, limit to max size
+                    if r > max_radius:
+                        r = max_radius
+                    # store in list of holes
+                    pt4 = Point(x, y, r)
+                    pt_list.append(pt4)
+                    # and finally paint the hole
+                    if style == "round":
+                        self.hole(x, y, r=r)
+                    elif style == "triangle":
```

Comment:
May be just pre calculate the number of corners and the angle and then just `else: self.regularPolygonHole(x, y, r=r, n=n, a=a)`

Reply:
Alternatively allow n=0 for getting a circle.

Reply:
Done.